app/code/core/Mage/Core/Model/Email/Queue.php

```diff
@@ -163,9 +163,41 @@ public function setRecipients(array $recipients)
      */
     public function getRecipients()
     {
+        if (empty($this->_recipients) && $this->getId()) {
+            // Load recipients from database if not already loaded
+            $this->_loadRecipients();
+        }
         return $this->_recipients;
     }
 
+    /**
+     * Load recipients from database
+     *
+     * @return $this
+     */
+    protected function _loadRecipients()
+    {
+        $resource = Mage::getSingleton('core/resource');
+        $connection = $resource->getConnection('core_read');
+        $table = $resource->getTableName('core/email_recipients');
+
+        $select = $connection->select()
+            ->from($table)
+            ->where('message_id = ?', $this->getId());
+
+        $rows = $connection->fetchAll($select);
+
+        foreach ($rows as $row) {
+            $this->_recipients[] = [
+                $row['recipient_email'],
+                $row['recipient_name'] ?: '',
+                (int) $row['email_type'],
+            ];
+        }
+
+        return $this;
+    }
```

Comment:
Only just looking at this on my phone, but I'd move this into a resource model since ideally regular models don't can the DB directly.

This table is only relevant for queued emails, right? So `Mage_Core_Model_Resource_Email_Queue` would seem correct, which already has similar code anyway.